airbyte-webapp/src/components/connection/UpdateConnectionDataResidency/UpdateConnectionDataResidency.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { ControlLabels } from "components/LabeledControl";
+import { Card } from "components/ui/Card";
+import { DropDown } from "components/ui/DropDown";
+import { Spinner } from "components/ui/Spinner";
+
+import { Geography } from "core/request/AirbyteClient";
+import { useConnectionEditService } from "hooks/services/ConnectionEdit/ConnectionEditService";
+import { useNotificationService } from "hooks/services/Notification";
+import { useAvailableGeographies } from "packages/cloud/services/geographies/GeographiesService";
+import { links } from "utils/links";
+
+import styles from "./UpdateConnectionDataResidency.module.scss";
+
+export const UpdateConnectionDataResidency: React.FC = () => {
+  const { connection, updateConnection, connectionUpdating } = useConnectionEditService();
+  const { registerNotification } = useNotificationService();
+  const { formatMessage } = useIntl();
+  const [selectedValue, setSelectedValue] = useState<Geography>();
+
+  const { geographies } = useAvailableGeographies();
+
+  const handleSubmit = async ({ value }: { value: Geography }) => {
+    try {
+      setSelectedValue(value);
+      await updateConnection({
+        connectionId: connection.connectionId,
+        geography: value,
+      });
+    } catch (e) {
+      registerNotification({
+        id: "connection.geographyUpdateError",
+        title: formatMessage({ id: "connection.geographyUpdateError" }),
+        isError: true,
+      });
+    }
+    setSelectedValue(undefined);
+  };
+
+  return (
+    <Card withPadding>
+      <div className={styles.wrapper}>
+        <div>
+          <ControlLabels
+            nextLine
+            label={<FormattedMessage id="connection.geographyTitle" />}
+            message={
+              <FormattedMessage
```

Comment:
Would it be worth to point here out, when this will take effect? I assume it's not cancelling the current running sync and I also assume it doesn't require a reset, so simply the next sync/reset jobs will be done in that region? Maybe worth putting something like: "This will take effect for all syncs and resets starting after the change." (or something better worded).

Reply:
I think this is almost self-explanatory and not necessary to explain, but maybe @andyjih can weigh in. It's also something we could consider adding to the docs instead of directly in the UI.

Here's a proposal for the wording if we do want to mention it:

> Depending on your network configuration, you may need to \<lnk\>add IP addresses\</lnk\> to your allowlist. Changes to data residency will be applied to new sync and reset jobs. Currently running jobs will not be affected.